extensions/annotations/src/main/java/io/opentelemetry/extension/annotations/TimerAttribute.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks that a parameter of a method or constructor annotated by the {@link Timed}
+ * annotation should be added as an attribute to the instrument.
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that the attribute should be created.
+ *
+ * <p>If you are a library developer, then probably you should NOT use this annotation, because it
+ * is non-functional without the OpenTelemetry auto-instrumentation agent, or some other annotation
+ * processor.
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface TimerAttribute {
```

Comment:
Could we perhaps reuse `SpanAttribute` somehow? Or deprecate it, and introduce a new annotation that would apply to both signals?

Reply:
@jkwatson true, it does not make sense to use both `@Timed` and `@Counted` since timers already include the count. I'm happy to merge `TimerAttribute` and `CounterAttribute` into a single `MetricAttribute`.